tools/golang-json-rpc-tests/main.go

```diff
@@ -0,0 +1,412 @@
+    err := rpcClient.CallContext(context.Background(), &result, "eth_accounts")
+    if err != nil {
+        log.Fatalf("Failed to get accounts: %v", err)
+    }
+
+    for _, account := range result {
+        address := common.HexToAddress(account)
+        balance, err := client.BalanceAt(context.Background(), address, nil)
+        if (err != nil) {
+            fmt.Printf("Failed to get balance for account %s: %v\n", account, err)
+            continue
+        }
+        if balance == nil {
+            log.Fatalf("Balance for account %s is nil", account)
+        }
+        fmt.Printf("Account %s balance: %s\n", account, balance.String())
+    }
+}
+
+func testFeeHistory(client *ethclient.Client, blockCount uint64, newestBlock *big.Int, rewardPercentiles []float64) {
+    ctx := context.Background()
+
+    feeHistory, err := client.FeeHistory(ctx, blockCount, newestBlock, rewardPercentiles)
+    if err != nil {
+        log.Fatalf("Failed to get fee history: %v", err)
+    }
+
+    if feeHistory == nil {
+        log.Fatalf("Fee history is nil")
+    }
+
+    fmt.Printf("Fee History:\n")
+    fmt.Printf("Oldest Block: %s\n", feeHistory.OldestBlock.String())
+    fmt.Printf("Gas Used Ratio: %v\n", feeHistory.GasUsedRatio)
+    fmt.Printf("Reward: %v\n", feeHistory.Reward)
+}
+
+func waitForTransaction(client *ethclient.Client, tx *types.Transaction) *types.Receipt {
+    ctx := context.Background()
+    receipt, err := bind.WaitMined(ctx, client, tx)
+    if err != nil {
+        log.Fatalf("Failed to wait for transaction to be mined: %v", err)
+    }
+    if receipt == nil {
+        log.Fatalf("Receipt is nil")
+    }
+    return receipt
+}
+
+func testGetBlockTransactionCountByHash(client *ethclient.Client, blockHash common.Hash) {
+    txCount, err := client.TransactionCount(context.Background(), blockHash)
+    if err != nil {
+        log.Fatalf("Failed to get transaction count by block hash: %v", err)
+    }
+    if txCount < 0 {
+        log.Fatalf("Transaction count is invalid: %d", txCount)
+    }
+    fmt.Printf("Transaction count by block hash %s: %d\n", blockHash.Hex(), txCount)
+}
+
+func testGetBlockTransactionCountByNumber(client *ethclient.Client) {
+    txCount, err := client.PendingTransactionCount(context.Background())
+    if err != nil {
+        log.Fatalf("Failed to get transaction count by block number: %v", err)
+    }
+    if txCount < 0 {
+        log.Fatalf("Transaction count is invalid: %d", txCount)
+    }
+    fmt.Printf("Transaction count by block number: %d\n", txCount)
+}
+
+func testCodeAt(client *ethclient.Client, address common.Address) {
+    code, err := client.CodeAt(context.Background(), address, nil)
+    if err != nil {
+        log.Fatalf("Failed to get code at address: %v", err)
+    }
+    if code == nil {
+        log.Fatalf("Code is nil")
+    }
+    fmt.Printf("Code at address %s: %s\n", address.Hex(), hex.EncodeToString(code)) // Not a SmartContract so empty string expected
```

Comment:
If we decide to add coverage for `eth_sendRawTransaction` to deploy an SC, we can use it here so .CodeAt can return something other than empty string.

Reply:
This code is no longer empty because of: https://github.com/hashgraph/hedera-json-rpc-relay/pull/2586#discussion_r1650705127